usecases/permissions_requests.go

```diff
@@ -0,0 +1,120 @@
+package usecases
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/topfreegames/Will.IAM/models"
+	"github.com/topfreegames/Will.IAM/repositories"
+)
+
+// PermissionsRequests define entrypoints for PermissionsRequests actions
+type PermissionsRequests interface {
+	Create(*models.PermissionRequest) error
+	Deny(string, string) error
+	Grant(string, string) error
+	ListOpenRequestsVisibleTo(
+		*repositories.ListOptions, string,
+	) ([]models.PermissionRequest, int64, error)
+	WithContext(context.Context) PermissionsRequests
+}
+
+type permissionsRequests struct {
+	repo *repositories.All
+	ctx  context.Context
+}
+
+func (prs permissionsRequests) WithContext(ctx context.Context) PermissionsRequests {
+	return &permissionsRequests{prs.repo.WithContext(ctx), ctx}
+}
+
+// Create checks if pr.saID has open request OR has permission, if not it opens a permission request
+func (prs permissionsRequests) Create(pr *models.PermissionRequest) error {
+	return prs.repo.WithPGTx(prs.ctx, func(repo *repositories.All) error {
+		pr.State = models.PermissionRequestStates.Open
+		has, err := repo.ServiceAccounts.HasPermission(pr.ServiceAccountID, pr.Permission())
```

Comment:
Suggestion: I'd go with a switch here.

```golang
switch has, err := ...; {
case err !=nil:
  return err
case has:
  return fmt.Errorf("...")
}
```

Reply:
oh wow, had no idea this was even a thing. I like it

Reply:
Great :)

Switches in Go are rather powerful and an idiomatic alternative over if/else if chains.